OpenRA.Mods.Common/Widgets/Logic/MapChooserLogic.cs

```diff
@@ -126,6 +126,18 @@ public class MapChooserLogic : ChromeLogic
 			SetupMapTab(MapClassification.User, filter, "USER_MAPS_TAB_BUTTON", "USER_MAPS_TAB", itemTemplate);
 			SetupMapTab(MapClassification.System, filter, "SYSTEM_MAPS_TAB_BUTTON", "SYSTEM_MAPS_TAB", itemTemplate);
 
+			if (modData.MapCache.LastUpdatedUid != null)
+			{
+				selectedUid = modData.MapCache.LastUpdatedUid;
+				currentTab = tabMaps.Keys.FirstOrDefault(k => tabMaps[k].Select(mp => mp.Uid).Contains(selectedUid));
+
+				if (currentTab != MapClassification.Unknown)
+				{
+					SwitchTab(currentTab, itemTemplate);
+					return;
+				}
+			}
```

Comment:
Having a `LastUpdatedUid` feels really inelegant. Can you explain to me what this logic is trying to achieve?

Reply:
I agree, it was even pretty buggy. I moved it to `LobbyLogic` 
It now relies on `MapUpdated` event, but inclusion of it made me make `UpdateMaps` public again. I didn't find another non-hacky solution to trigger an update

Reply:
Oh and for the explanation. The goal of it is when a user first enters map chooser, the last updated map is preselected. In basically all scenarios where a user updates or adds a map he wants to immediately chose it in the lobby. So this is a shortcut. 
Though it doesn't work if a user is outside the game lobby as `LobbyLogic` won't be subscribed to updates

Reply:
`UpdateMaps` being public is fine, it seems reasonable to ask a cache to refresh itself.

Much prefer the new logic flow you've created here, it's much nicer that the `LobbyLogic` is tracking the last updated ID for it's own purposes rather than relying on another class to do it.